core/src/main/java/io/questdb/griffin/engine/functions/catalogue/ShowTablesFunctionFactory.java

```diff
@@ -176,55 +177,50 @@ private void of(SqlExecutionContext executionContext) {
             }
 
             private class TableListRecord implements Record {
-                private boolean isDedup;
-                private boolean isSoftLink;
-                private int maxUncommittedRows;
-                private long o3MaxLag;
-                private int partitionBy;
-                private CharSequence timestampColumnName;
+                private CairoTable table = new CairoTable();
 
                 @Override
                 public boolean getBool(int col) {
                     if (col == WAL_ENABLED_COLUMN) {
-                        return tableToken.isWal();
+                        return table.getWalEnabled();
                     }
                     if (col == DEDUP_NAME_COLUMN) {
-                        return isDedup;
+                        return table.getIsDedup();
                     }
                     return false;
                 }
 
                 @Override
                 public int getInt(int col) {
                     if (col == ID_COLUMN) {
-                        return tableToken.getTableId();
+                        return table.getId();
                     }
                     assert col == MAX_UNCOMMITTED_ROWS_COLUMN;
-                    return maxUncommittedRows;
+                    return table.getMaxUncommittedRows();
                 }
 
                 @Override
                 public long getLong(int col) {
                     assert col == O3_MAX_LAG_COLUMN;
-                    return o3MaxLag;
+                    return table.getO3MaxLag();
                 }
 
                 @Override
                 public CharSequence getStrA(int col) {
                     if (Chars.equals(ShowTablesCursorFactory.TABLE_NAME_COLUMN_NAME, getMetadata().getColumnName(col))) {
-                        return tableToken.getTableName();
+                        return table.getName();
```

Comment:
the line above this: this is not your code but it is such a convoluted way to find out if we should produce table name. This whole method should probably be a switch statement